src/webapp/ldap_data.py

```diff
@@ -158,16 +158,17 @@ def get_ligo_ldap_dns(ldap_url: str, ldapuser: str, ldap_pass: str) -> List[str]
         log.exception("Failed to connect to the LIGO LDAP")
         return results
 
-    for group in ('people', 'robot'):
-        try:
-            conn.search(base_branch.format(group=group),
-                        queries[group],
-                        search_scope='SUBTREE',
-                        attributes=['gridX509subject'])
-            results += [dn for e in conn.entries for dn in e.gridX509subject]
-        except ldap3.core.exceptions.LDAPException:
-            log.exception("Failed to query LIGO LDAP for %s DNs", group)
-
-    conn.unbind()
+    try:
+        for group in ('people', 'robot'):
+            try:
+                conn.search(base_branch.format(group=group),
+                            queries[group],
+                            search_scope='SUBTREE',
+                            attributes=['gridX509subject'])
+                results += [dn for e in conn.entries for dn in e.gridX509subject]
+            except ldap3.core.exceptions.LDAPException:
+                log.exception("Failed to query LIGO LDAP for %s DNs", group)
+    finally:
+        conn.unbind()
```

Comment:
i squinted at this for a while and then gave up and did a 

```git show --color-words 46e1faff```

and suddenly it all made sense.

Too bad github was not very good at highlighting the actual change here (inserting `try:` and `finally:`)